runtime/autoload/man.vim

```diff
@@ -242,18 +243,26 @@ function! man#complete(arg_lead, cmd_line, cursor_pos) abort
     let name = a:arg_lead
     let sect = tolower(args[1])
   endif
-  " We remove duplicates incase the same manpage in different languages was found.
-  return uniq(sort(map(globpath(s:mandirs,'man?/'.name.'*.'.sect.'*', 0, 1), 's:format_candidate(v:val, sect)'), 'i'))
+  return s:complete(sect, sect, name)
 endfunction
 
-function! s:format_candidate(path, sect) abort
+function! s:complete(sect, psect, name) abort
+  let old_fic = &fileignorecase
+  let &fileignorecase = &wildignorecase
```

Comment:
While `'fileignorecase'` is what actually controls the behaviour, `'wildignorecase'` is more applicable. It is supposed to control the case insensitivity of completing, whereas setting `'fileignorecase'` will apply everywhere. Personally, I set `'wildignorecase'` for case insensitive completion, but I don't want files or directories to be globally treated with case insensitivity.

Also, I noticed a mistake, if `'fileignorecase'` is already set, then we shouldn't be looking at the value of `'wildignorecase'` since according to `:h wildignorecase`, `'wildignorecase'` should do nothing if `'fileignorecase'` is set.

Shall I create a PR to revert this or fix it?